### PR TITLE
Deletes product from cart

### PR DESCRIPTION
**Pull request by trc311**
Closes Issue #20

### CODE REVIEW

File: shopcart.py [hunk 1]
```diff
@@ -237,6 +237,19 @@ def create_products(sid):
...
+@app.route('/shopcarts/<int:sid>/products/<int:sku>', methods=['DELETE'])
+def delete_products(sid,sku):
+    for i in range(len(shopping_carts)):
+    	if shopping_carts[i]['sid'] == sid:
+    	    for j in range(len(shopping_carts[i]['products'])):
+    	    	if shopping_carts[i]['products'][j]['sku'] == sku:
+    	    		del shopping_carts[i]['products'][j]
+    	    		break
+    return '', HTTP_204_NO_CONTENT
+
```
Comment (Pratik-Karnik) [inline shopcart.py]: I think we should show appropriate error messages of cart not present and product not present if user enters invalid sid or sku. What do you think?

**Comment by Pratik-Karnik** [comment]
@senongtor @WesPainter Have you guys reviewed this PR? Are you guys also okay with these changes? If so then I will merge it.

File: shopcart.py (hunk 1 above)
Comment (senongtor) [reply shopcart.py]: I think it should split the error msg to 'shop cart not found' and 'product not found'

Comment (trc311) [reply shopcart.py]: If you do a get after you delete, it gives that message

**Comment by WesPainter** [comment]
The implementation itself looks okay, but I'm not sure how to resolve the conflicts.  It looks like its choosing to keep either the delete products, or delete carts.  Any ideas?

**Comment by senongtor** [comment]
I think we should stick to DELETEPRODUCT not MASTER 
@WesPainter @trc311

**Comment by Pratik-Karnik** [comment]
Even if the product or the shopping cart does not exist we still have to show that it is deleted which I feel this implementation does.

**Comment by WesPainter** [comment]
I am talking about conflicts that must be resolved to merge the branch. @trc311 Maybe what you can do is merge the master branches with your own locally, then commit those changes to your branch.  That should resolve the conflicts.

**Comment by Pratik-Karnik** [comment]
Or maybe he can pull the latest from Master and resolve conflicts on his local and push back the new changes.

**Comment by senongtor** [comment]
@trc311 you can also checkout master, pull and then checkout your branch and do rebase on master. Then manually resolve the conflict and force push

**Comment by trc311** [comment]
Resolved conflicts, the code is running fine, tested it as well

**Pull request merged**